src/main/java/io/github/jamsesso/jsonlogic/evaluator/expressions/DurationArithmeticExpression.java

```diff
@@ -0,0 +1,86 @@
+package io.github.jamsesso.jsonlogic.evaluator.expressions;
+
+import io.github.jamsesso.jsonlogic.evaluator.JsonLogicEvaluationException;
+import io.github.jamsesso.jsonlogic.utils.DateOperations;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Objects;
+
+public class DurationArithmeticExpression implements PreEvaluatedArgumentsExpression {
+  public static final DurationArithmeticExpression DURATION_OF = new DurationArithmeticExpression("duration.of");
+  public static final DurationArithmeticExpression DURATION_BETWEEN = new DurationArithmeticExpression("duration.between");
+  public static final DurationArithmeticExpression DURATION_AS = new DurationArithmeticExpression("duration.as");
+
+  private final String key;
+
+  public DurationArithmeticExpression(String key) {
+    this.key = key;
+  }
+
+  @Override
+  public String key() {
+    return key;
+  }
+
+  @Override
+  public Object evaluate(List arguments, Object data) throws JsonLogicEvaluationException {
+    if (arguments.isEmpty()) {
+      return null;
+    }
+
+    if (arguments.size() != 2) {
+      throw new JsonLogicEvaluationException("duration arithmetic requires 2 arguments");
+    }
+
+    /*
+     * --- Supported Operations ---
+     * DURATION_OF ("duration_of")
+     *    1. [intervalToAdd, intervalUnit]
+     * DURATION_BETWEEN ("duration_between")
+     *    1. [Date, Date]
+     *
+     * RETURNS: Duration as String
+     *
+     * DURATION_AS
+     *    1. [Duration, intervalUnit]
+     *
+     * RETURNS: interval as NUMBER (BigDecimal)
+     */
+
+    try {
+      if (Objects.equals(key, "duration.between")) {
+        OffsetDateTime start = DateOperations.fromDateString((String) arguments.get(0));
+        OffsetDateTime end = DateOperations.fromDateString((String) arguments.get(1));
+        return Duration.between(start, end).toString();
+      } else if (Objects.equals(key, "duration.of")) {
+        BigDecimal intervalToAdd = (BigDecimal) arguments.get(0);
+        ChronoUnit intervalUnit = ChronoUnit.valueOf((String) arguments.get(1));
+        return Duration.of(Long.parseLong(intervalToAdd.toString()), intervalUnit).toString();
+      } else {
+//        key == duration.as
+        Duration duration = Duration.parse((String) arguments.get(0));
+        ChronoUnit intervalUnit = ChronoUnit.valueOf((String) arguments.get(1));
+        switch (intervalUnit) {
```

Comment:
Add months and years as well. 
Note that ISO8601 also supports fractions.
From: https://en.wikipedia.org/wiki/ISO_8601
<img width="958" alt="image" src="https://github.com/TogaiHQ/json-logic-java/assets/45751032/babb7665-cf54-429e-8d35-dd922a6ea978">